### ISSUE TITLE
Heroku and Kaleido

### ISSUE THREAD

**Issue by MeganCowie** [plotly/Kaleido#73]
I was trying to deploy an app to Heroku today and Kaleido kept getting in the way, with the error: 
ERROR: Could not find a version that satisfies the requirement kaleido==0.0.1 ERROR: No matching distribution found for kaleido==0.0.1

I was able to get around it by deleting Kaleido from my requirements.txt file so it's not a pressing issue for me, but I wanted to let you know in case it's relevant for others / future.  

ps - I tried updating both pip and kaleido, and these ideas did not solve the issue.

**Comment by jonmmease** [comment]
Hi @MeganCowie, what was the entry you had in `requirements.txt` while receiving the error? Was there any version constraint on it?

**Comment by MeganCowie** [comment]
Hi Jon, Wow fast reply thanks :) Yes, the line is: "kaleido==0.1.0.post1", which is the result of creating a requirements.txt file via 
$ pip freeze > requirements.txt

**Comment by MeganCowie** [comment]
Hi Jon, 

I just recreated the issue, and here are some more details in case it's helpful: 

1) requirements.txt file has the line kaleido==0.1.0.post1
2) After committing this update to git I try to deploy to Heroku via $ git push heroku main
3) This is the complete message:

Enumerating objects: 13, done.
Counting objects: 100% (13/13), done.
Delta compression using up to 12 threads
Compressing objects: 100% (12/12), done.
Writing objects: 100% (13/13), 3.16 KiB | 1.05 MiB/s, done.
Total 13 (delta 3), reused 0 (delta 0), pack-reused 0
remote: Compressing source files... done.
remote: Building source:
remote:
remote: -----> Building on the Heroku-20 stack
remote: -----> Python app detected
remote: -----> Installing python-3.6.13
remote: -----> Installing pip 20.1.1, setuptools 47.1.1 and wheel 0.34.2
remote: -----> Installing SQLite3
remote: -----> Installing requirements with pip
remote:        Collecting appdirs==1.4.4
remote:          Downloading appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
remote:        Collecting asgiref==3.2.7
remote:          Downloading asgiref-3.2.7-py2.py3-none-any.whl (19 kB)
remote:        Collecting Brotli==1.0.7
remote:          Downloading Brotli-1.0.7-cp36-cp36m-manylinux1_x86_64.whl (352 kB)
remote:        Collecting certifi==2020.6.20
remote:          Downloading certifi-2020.6.20-py2.py3-none-any.whl (156 kB)
remote:        Collecting chardet==3.0.4
remote:          Downloading chardet-3.0.4-py2.py3-none-any.whl (133 kB)
remote:        Collecting click==7.1.2
remote:          Downloading click-7.1.2-py2.py3-none-any.whl (82 kB)
remote:        Collecting dash==1.13.3
remote:          Downloading dash-1.13.3.tar.gz (67 kB)
remote:        Collecting dash-bootstrap-components==0.10.3
remote:          Downloading dash-bootstrap-components-0.10.3.tar.gz (107 kB)
remote:          Installing build dependencies: started
remote:          Installing build dependencies: finished with status 'done'
remote:          Getting requirements to build wheel: started
remote:          Getting requirements to build wheel: finished with status 'done'
remote:            Preparing wheel metadata: started
remote:            Preparing wheel metadata: finished with status 'done'
remote:        Collecting dash-core-components==1.10.1
remote:          Downloading dash_core_components-1.10.1.tar.gz (3.4 MB)
remote:        Collecting dash-daq==0.5.0
remote:          Downloading dash_daq-0.5.0.tar.gz (642 kB)
remote:        Collecting dash-html-components==1.0.3
remote:          Downloading dash_html_components-1.0.3.tar.gz (187 kB)
remote:        Collecting dash-renderer==1.5.0
remote:          Downloading dash_renderer-1.5.0.tar.gz (1.2 MB)
remote:        Collecting dash-table==4.8.1
remote:          Downloading dash_table-4.8.1.tar.gz (1.8 MB)
remote:        Collecting datedelta==1.3
remote:          Downloading datedelta-1.3-py3-none-any.whl (6.7 kB)
remote:        Collecting distlib==0.3.1
remote:          Downloading distlib-0.3.1-py2.py3-none-any.whl (335 kB)
remote:        Collecting django==3.0.5
remote:          Downloading Django-3.0.5-py3-none-any.whl (7.5 MB)
remote:        Collecting filelock==3.0.12
remote:          Downloading filelock-3.0.12-py3-none-any.whl (7.6 kB)
remote:        Collecting Flask==1.1.2
remote:          Downloading Flask-1.1.2-py2.py3-none-any.whl (94 kB)
remote:        Collecting Flask-Compress==1.5.0
remote:          Downloading Flask-Compress-1.5.0.tar.gz (6.5 kB)
remote:        Collecting future==0.18.2
remote:          Downloading future-0.18.2.tar.gz (829 kB)
remote:        Collecting gunicorn==20.0.4
remote:          Downloading gunicorn-20.0.4-py2.py3-none-any.whl (77 kB)
remote:        Collecting idna==2.10
remote:          Downloading idna-2.10-py2.py3-none-any.whl (58 kB)
remote:        Collecting itsdangerous==1.1.0
remote:          Downloading itsdangerous-1.1.0-py2.py3-none-any.whl (16 kB)
remote:        Collecting Jinja2==2.11.2
remote:          Downloading Jinja2-2.11.2-py2.py3-none-any.whl (125 kB)
remote:        Collecting joblib==0.16.0
remote:          Downloading joblib-0.16.0-py3-none-any.whl (300 kB)
**remote:        ERROR: Could not find a version that satisfies the requirement kaleido==0.1.0.post1 (from -r /tmp/build_8cfc62cd/requirements.txt (line 26)) (from versions: 0.0.1rc3, 0.0.1rc4, 0.0.1rc5, 0.0.1rc6, 0.0.1rc8, 0.0.1rc9, 0.0.1, 0.0.2, 0.0.3, 0.0.3.post1, 0.1.0a2, 0.1.0a3, 0.1.0)
remote:        ERROR: No matching distribution found for kaleido==0.1.0.post1 (from -r /tmp/build_8cfc62cd/requirements.txt (line 26))**
remote:  !     Push rejected, failed to compile Python app.
remote:
remote:  !     Push failed
remote: Verifying deploy...
remote:
remote: !       Push rejected to templatedashapp.
remote:
To https://git.heroku.com/templatedashapp.git
 ! [remote rejected] main -> main (pre-receive hook declined)
error: failed to push some refs to 'https://git.heroku.com/templatedashapp.git'

4) I checked my installed version of Kaleido via $ pip show kaleido which returns:

Name: kaleido
Version: 0.1.0.post1
Summary: Static image export for web-based visualization libraries with zero dependencies
Home-page: UNKNOWN
Author: Jon Mease
Author-email: jon@plotly.com
License: MIT
Location: c:\users\m-cow\appdata\local\programs\python\python38-32\lib\site-packages
Requires:
Required-by:

5) I have tried uninstalling and reinstalling pip and kaleido according to your discussion in this issue https://github.com/plotly/Kaleido/issues/15
but the problem persisted. 


Best,
Megan
